### PR TITLE
Added ControlsTimer class with example usage in mazda+hyundai

### PR DESCRIPTION
**Pull request by MoreTore**
Increase the readability of DT_CTRL based duration timers in carports. Also, remove redundant self.frame+=1 in every carcontroller. See examples for Mazda and Hyundai.

### CODE REVIEW

**Comment by MoreTore** [comment]
Not sure why process replay is failing. The code should be good.

File: common/realtime.py [hunk 1]
```diff
@@ -94,3 +94,89 @@ def monitor_time(self) -> bool:
...
+
+  def once_after_reset(self) -> bool: 
+    """Returns true only one time after calling reset()"""
+    ret = self.was_reset
+    self.was_reset = False
+    return ret
+
+  @staticmethod
+  def interval_obj(rate, frame) -> bool:
+    if frame % rate == 0: # Highlighting shows "frame" in white
```
Comment (MoreTore) [inline common/realtime.py:134]: Forgot to remove the comment

**Comment by adeebshihadeh** [comment]
I like the direction this is going in. Gonna move to draft until I have some time to look at it more.

**Comment by MoreTore** [comment]
Let me know if you have any questions about it

**Comment by MoreTore** [comment]
I added some unit tests in case you make changes

**Pull request closed without merging**